proto/regen/ecocredit/v1alpha1/types.proto

```diff
@@ -117,4 +117,100 @@ message CreditTypeSeq {
 
   // The sequence number of classes of the credit type
   uint64 seq_number = 2;
+}
+
+// SellOrder is the content of a new sell order.
+message SellOrder {
+
+  // address is the address of the seller
+  optional string address = 1;
+
+  // sell_order_id is the id of the sell order
+  optional uint64 sell_order_id = 2;
+
+  // batch_denom is the credit batch being sold.
+  string batch_denom = 3;
+
+  // quantity is the quantity of credits being sold from this batch. If it is
+  // less then the balance of credits the owner has available at the time this
+  // sell order is matched, the quantity will be adjusted downwards to the
+  // owner's balance. However, if the balance of credits is less than this
+  // quantity at the time the sell order is created, the operation will fail.
+  string quantity = 4;
+
+  // ask_price is the price the seller is asking for each unit of the
+  // batch_denom. Each credit unit of the batch will be sold for at least the
+  // ask_price or more.
+  cosmos.base.v1beta1.Coin ask_price = 5;
+
+  // disable_auto_retire disables auto-retirement of credits which allows a
+  // buyer to disable auto-retirement in their buy order enabling them to
+  // resell the credits to another buyer.
+  bool disable_auto_retire = 6;
+}
+
+// BuyOrder is the content of a new buy order.
+message BuyOrder {
+
+  // address is the address of the buyer
+  optional string address = 1;
+
+  // buy_order_id is the id of the buy order
+  optional uint64 buy_order_id = 2;
```

Comment:
not sure if this is the best way to do this. wanted to avoid duplicating BuyOrder and SellOrder messages for query/tx types. Doing this allows us to reuse and optionally fill in these values when its used in a query. When used as a tx, there is only one address at the top level used as the signer. 

Another option is to remove these optionals, and include an additional repeated string and repeated uint64 who's index will match that of the repeated sell/buy orders in queries.

Reply:
Why would we make `address` and `buy_order_id` optional? These should both be required in the transaction message and included in the query response.

Reply:
so address was made optional as the `MsgBuy` structure has the buyer address at the top level, and not in the `Order` message. I wanted to reuse the `Order` structure in the query, while being able to inject address for query purposes. 

perhaps the better way is either to make a new order structure for queries or reuse the old one and just have another repeated field who's index matches the index of the `Order` msg

Reply:
I think we should avoid making them optional for the purpose of reusing code. I think each of the query responses should return `SellOrder` and `BuyOrder` as they are defined in `types.proto` (`BuyOrder` should include the buyer address, which I overlooked in #613) and the `Orders` within `MsgSell` and `MsgBuy` should stay the same.